RevenueCatUI/View+PresentPaywall.swift

```diff
@@ -140,7 +150,7 @@ private struct PresentingPaywallModifier: ViewModifier {
 
     func body(content: Content) -> some View {
         content
-            .sheet(item: self.$data) { data in
+            .sheet(item: self.$data, onDismiss: self.onDismiss) { data in
```

Comment:
I tested this and it gets called when closing the sheet manually by the user and programatically.

Unfortunately I wasn't able to write a unit test that gets called because of the way we hack adding SwiftUI views to the hierarchy without a host app in the test target.